hal/armv7a/cpu.c

```diff
@@ -199,3 +200,15 @@ char *hal_cpuFeatures(char *features, unsigned int len)
 
 	return features;
 }
+
+
+void hal_cpuTlsSet(tls_t *tls, cpu_context_t *ctx)
+{
+	/* In theory there should be 8-byte thread control block but
+	 * it's stored elsewhere so we need to subtract 8 from the pointer
+	 */
+	u32 ptr = (u32)tls->tls_base - 8;
```

Comment:
Why is it stored in a different place than expected? This might be covering some hidden bug elsewhere.
Also `ptr_t` should be used in place of `u32`.

Reply:
Ok, maybe I've misunderstood something, but GCC generates calls for first TLS variable 8 bytes over the address in the register. ARM abi only specifies TLS for Linux.